src/main/java/com/bean00/server/RequestProcessor.java

```diff
@@ -21,34 +34,61 @@ public Response processRequest(Request request) throws IOException {
         String requestURL = request.getRequestURL();
 
         Response response;
-        if (requestURL.equals("/")) {
-            response = new Response(Status.OK);
-        } else if (!dataStore.fileExists(requestURL)) {
-            response = new Response(Status.NOT_FOUND);
-        } else if (requestMethod.equals("HEAD")) {
-            byte[] body = dataStore.getData(requestURL);
+        if (dataStore.isDirectory(requestURL)) {
+            byte[] body = buildHtmlAsAByteArray();
             HashMap<String, String> headers = buildHeaders(body, requestURL);
-            response = new Response(Status.OK, Method.HEAD, headers, body);
-        } else {
+            response = new Response(Status.OK, requestMethod, headers, body);
+        } else if (dataStore.isFile(requestURL)) {
             byte[] body = dataStore.getData(requestURL);
             HashMap<String, String> headers = buildHeaders(body, requestURL);
-            response = new Response(Status.OK, Method.GET, headers, body);
+            response = new Response(Status.OK, requestMethod, headers, body);
+        } else {
+            response = new Response(Status.NOT_FOUND);
         }
 
         return response;
     }
 
-    private HashMap<String, String> buildHeaders(byte[] body, String requestURL) throws IOException {
-        HashMap<String, String> headers = new HashMap<>();
+    private byte[] buildHtmlAsAByteArray() {
+        String htmlAsString = htmlBefore;
+        htmlAsString += getHtmlForListOfContents();
+        htmlAsString += htmlAfter;
+        byte[] body = htmlAsString.getBytes();
 
-        int contentLength = body.length;
-        String contentLengthAsString = Integer.toString(contentLength);
-        headers.put("Content-Length", contentLengthAsString);
+        return body;
+    }
+
+    private String getHtmlForListOfContents() {
```

Comment:
It's starting to look like you're giving this class more responsibilities than it should take care of

Reply:
I've addressed this